pkg/component/runtime/manager.go

```diff
@@ -192,7 +191,7 @@ func NewManager(
 		errCh:          make(chan error),
 		monitor:        monitor,
 		grpcConfig:     grpcConfig,
-		serverReady:    &atomic.Bool{},
+		serverReady:    make(chan struct{}, 1),
```

Comment:
Why does this need to be a buffered channel? 
If it's only using as a signal for when we entered `serverLoop`, a boolean seemed more appropriate to represent a running state instead of something that can be read only once.

Moreover if it's only used in testing, isn't there a better way to know if the manager itself is running without always allocating a channel and always writing to it also for production code?

Reply:
@michel-laterman @pchila has a point about being able to check it once. This change does prevent the `waitForReady` from being able to be called more than once.

Reply:
I'll change to closing an unbuffered channel on ready; this is the same approach we take in fleet-server with the policy monitor